### PR TITLE
200911 Billing adjustments

### LINKED ISSUE #14: Stripe webhook failing due to CSRF
It looks like some configuration changed as all webhooks from Stripe are failing due to a CSRF error, see below.

```
<!DOCTYPE html>
<html lang="en">
<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta name="robots" content="NONE,NOARCHIVE">
  <title>403 Forbidden</title>
  <style type="text/css">
    html * { padding:0; margin:0; }
    body * { padding:10px 20px; }
    body * * { padding:0; }
    body { font:small sans-serif; background:#eee; color:#000; }
    body>div { border-bottom:1px solid #ddd; }
    h1 { font-weight:normal; margin-bottom:.4em; }
    h1 span { font-size:60%; color:#666; font-weight:normal; }
    #info { background:#f6f6f6; }
    #info ul { margin: 0.5em 4em; }
    #info p, #summary p { padding-top:10px; }
    #summary { background: #ffc; }
    #explanation { background:#eee; border-bottom: 0px none; }
  </style>
</head>
<body>
<div id="summary">
  <h1>Forbidden <span>(403)</span></h1>
  <p>CSRF verification failed. Request aborted.</p>
  <p>You are seeing this message because this HTTPS site requires a “Referer header” to be sent by your Web browser, but none was sent. This header is required for security reasons, to ensure that your browser is not being hijacked by third parties.</p>
  <p>If you have configured your browser to disable “Referer” headers, please re-enable them, at least for this site, or for HTTPS connections, or for “same-origin” requests.</p>
  <p>If you are using the &lt;meta name=&quot;referrer&quot; content=&quot;no-referrer&quot;&gt; tag or including the “Referrer-Policy: no-referrer” header, please remove them. The CSRF protection requires the “Referer” header to do strict referer checking. If you’re concerned about privacy, use alternatives like &lt;a rel=&quot;noreferrer&quot; …&gt; for links to third-party sites.</p>
</div>
<div id="explanation">
  <p><small>More information is available with DEBUG=True.</small></p>
</div>
</body>
</html>
```

### PR DESCRIPTION
**Pull request by paolodamico**
This PR introduces the following changes.

- Closes #14.
- Fixes a bug where the app relied on the `SITE_URL` env var for Stripe's return URL (which is not always set). The app now grabs the base domain from the request.
- Allows providing a custom message for each plan when the billing toolbar is shown (see below). **N.B. Requires handling on the main repo.**
screenshot
- Removes `STRIPE_DEFAULT_PRICE_ID` env var in favor of using the `Plan` model.
- The `/api/user` endpoint will now return the billing information when a plan is active for the team. Example:
```json
{
    "plan": {
        "key": "growth",
        "name": "Growth Plan",
        "custom_setup_billing_message": ""
    }
}
```
- Capability to use custom billing logic for some plans. This enables for instance the startup plan, which instead of creating a subscription only does a $0.50 USD pre-auth charge to the card that goes uncaptured.
- Handles `payment_intent.amount_capturable_updated` Stripe webhook to support only pre-auth charges for certain plans.
- Introduces an improvement to re-use already existing checkout sessions if they're still active, avoiding multiple unnecessary requests to Stripe that degrades the user load time and bloats Stripe with unnecessary data.
- Improves the welcome view to support multiple plans (instead of a static welcome to the Growth plan).
screenshot

### CODE REVIEW

File: multi_tenancy/templates/billing-welcome.html [hunk 1]
```diff
@@ -6,14 +6,21 @@
...
+          src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNCAyNCIgZmlsbD0iIzdBQzk1RSIgd2lkdGg9IjE0MHB4IiBoZWlnaHQ9IjE0MHB4Ij4KICAgICAgICAgIDxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiIC8+CiAgICAgICAgICA8cGF0aCBkPSJNMTIgMkM2LjQ4IDIgMiA2LjQ4IDIgMTJzNC40OCAxMCAxMCAxMCAxMC00LjQ4IDEwLTEwUzE3LjUyIDIgMTIgMnptLTIgMTVsLTUtNSAxLjQxLTEuNDFMMTAgMTQuMTdsNy41OS03LjU5TDE5IDhsLTkgOXoiLz4KPC9zdmc+"
       /></span>
-      <h1>Welcome to the PostHog <b>Growth Plan</b>!</h1>
+      <h1>You're all set!</h1>
+      {% if plan %}
       <h3>
-        You can now capture up to <b>500,000 events / month</b> to get better
-        insights into your users.
+        You are now subscribed to the <b>{{plan.name}}</b>. Please reach out to
```
Comment (Twixes) [inline multi_tenancy/templates/billing-welcome.html:14]: I think this should be `to the <b>{{plan.name}}</b> plan`?

**Review by Twixes — COMMENTED** [review]
Should be good! Could you just remove the new migrations and recreate them so that there's only one? On principle a PR to master should only introduce one migration, less bloat this way.

File: multi_tenancy/templates/billing-welcome.html (hunk 1 above)
Comment (paolodamico) [reply multi_tenancy/templates/billing-welcome.html:14]: Oh actually the plan names already include this, like "Startup Plan" or "Growth Plan".

**Comment by paolodamico** [comment]
> Should be good! Could you just remove the new migrations and recreate them so that there's only one? On principle a PR to master should only introduce one migration, less bloat this way.

Absolutely, my bad, forgot to squash them!

File: multi_tenancy/templates/billing-welcome.html (hunk 1 above)
Comment (Twixes) [reply multi_tenancy/templates/billing-welcome.html:14]: Ah, that's a bit weird tbh, but alright.

Comment (paolodamico) [reply multi_tenancy/templates/billing-welcome.html:14]: Yeah, the thing is I want the plan names to match on Stripe (to potentially create/sync them automatically), and if we don't add the "Plan" on the name shown on Stripe, it looks strange on the checkout / billing management processes.

**Pull request merged**